VideoRenderer/VideoRenderer/MediaCharacteristicRenderer.swift

```diff
@@ -0,0 +1,142 @@
+//  Copyright © 2016 One by Aol : Publishers. All rights reserved.
+
+import AVFoundation
+
+private let assetKey = "availableMediaCharacteristicsWithMediaSelectionOptions"
+
+public protocol UUIDPhantom: Hashable {
+    var uuid: UUID { get }
+}
+
+extension UUIDPhantom {
+    public var hashValue: Int { return uuid.hashValue }
+    public static func ==(left: Self, right: Self) -> Bool {
+        return left.uuid == right.uuid
+    }
+}
+
+func first<T, U>(pair: (T, U)) -> T { return pair.0 }
+func second<T, U>(pair: (T, U)) -> U { return pair.1 }
+
+public struct AvailableMediaOptions {
+    public struct Option: UUIDPhantom {
+        public let uuid: UUID
+        public let name: String
+        
+        public init(uuid: UUID = UUID(), name: String) {
+            self.uuid = uuid
+            self.name = name
+        }
+    }
+    
+    public static let empty = AvailableMediaOptions(unselectedOptions: [],
+                                                    selectedOption: nil)
+    public let unselectedOptions: [Option]
+    public let selectedOption: Option?
+}
+
+class MediaCharacteristicRenderer {
+    typealias Option = AvailableMediaOptions.Option
+    
+    struct Props {
+        let item: AVPlayerItem
+        let didStartMediaOptionsDiscovery: () -> ()
+        let didDiscoverAudibleOptions: (AvailableMediaOptions) -> ()
+        let didDiscoverLegibleOptions: (AvailableMediaOptions) -> ()
+        var selectedAudibleOption: Option?
+        var selectedLegibleOption: Option?
+    }
+    
+    struct MediaOptionCache {
```

Comment:
What is the purpose of this cache?

Reply:
It's cached available options for future selection